### PR TITLE
fix(kubernetes-probe): Fix inconstancy of probe file download + fix removing of the lock

### PR DESCRIPTION
**Pull request by Lowaiz**
Signed-off-by: Lyonel Martinez <lyonel.martinez@numberly.com>

**What type of PR is this?**

/kind bug

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

/area pkg

**What this PR does / why we need it**:

Fix inconstancy in probe module extraction. This bug was due to  a too slow compilation, letting Driverkit trying to get the file while it was still empty. Diverse fixes on lock behavior.

**Which issue(s) this PR fixes**:

None

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/driverbuilder/kubernetes.go [hunk 1]
```diff
@@ -270,16 +257,31 @@ func (bp *KubernetesBuildProcessor) copyModuleAndProbeFromPodWithUID(ctx context
...
+
 					err = copySingleFileFromPod(outModule, bp.coreV1Client, bp.clientConfig, p.Namespace, p.Name, builder.ModuleFullPath)
 					if err != nil {
 						return err
 					}
+					outModule.Close()
+					logger.Info("Kernel Module extraction successful")
 				}
 				if builder.ProbeFullPath != "" {
+					time.Sleep(15 * time.Second)
```
Comment (FedeDP) [inline pkg/driverbuilder/kubernetes.go:273]: Would it make sense to make this a config option?

File: pkg/driverbuilder/kubernetes.go [hunk 2]
```diff
@@ -270,16 +257,31 @@ func (bp *KubernetesBuildProcessor) copyModuleAndProbeFromPodWithUID(ctx context
 			if p.Status.Phase == corev1.PodRunning {
 				logger.WithField(falcoBuilderUIDLabel, falcoBuilderUID).Info("start downloading module and probe from pod")
 				if builder.ModuleFullPath != "" {
+					outModule, err := os.Create(build.ModuleFilePath)
+					if err != nil {
+						return err
+					}
+
```
Comment (FedeDP) [inline pkg/driverbuilder/kubernetes.go:264]: I'd defer the `Close()` here; you are leaking it if `copySingleFileFromPod` fails below.

File: pkg/driverbuilder/kubernetes.go [hunk 3]
```diff
@@ -270,16 +257,31 @@ func (bp *KubernetesBuildProcessor) copyModuleAndProbeFromPodWithUID(ctx context
...
 					}
+					outModule.Close()
+					logger.Info("Kernel Module extraction successful")
 				}
 				if builder.ProbeFullPath != "" {
+					time.Sleep(15 * time.Second)
+					outProbe, err := os.Create(build.ProbeFilePath)
+					if err != nil {
+						return err
+					}
```
Comment (FedeDP) [inline pkg/driverbuilder/kubernetes.go:277]: I'd defer the Close() here; you are leaking it if copySingleFileFromPod fails below.

File: pkg/driverbuilder/kubernetes.go [hunk 4]
```diff
@@ -296,14 +298,21 @@ func unlockPod(podClient v1.PodsGetter, clientConfig *restclient.Config, pod *co
 	options := &exec.ExecOptions{
 		PodClient: podClient,
 		Config:    clientConfig,
-		Pod:       pod,
+		StreamOptions: exec.StreamOptions{
+			IOStreams: genericclioptions.IOStreams{
+				Out:    bytes.NewBuffer([]byte{}),
+				ErrOut: bytes.NewBuffer([]byte{}), //TODO(fntlnz): necessary to deal with errors here?
```
Comment (FedeDP) [inline pkg/driverbuilder/kubernetes.go:304]: > TODO(fntlnz):

mmh? Copy pasted from another part of code? :D

File: pkg/driverbuilder/kubernetes.go (hunk 1 above)
Comment (Lowaiz) [reply pkg/driverbuilder/kubernetes.go]: I came up, this morning, with a new idea to handle this bug with file doing a lock role.
Now the downloader script wait until the provided lockfile is deleted to cat the compiled driver

**Review by FedeDP — APPROVED** [review]
/approve

**Pull request merged**